packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -210,46 +210,27 @@ class SliderThemeData extends Diagnosticable {
   /// ```
   /// {@end-tool}
   const SliderThemeData({
-    @required this.trackHeight,
-    @required this.activeTrackColor,
-    @required this.inactiveTrackColor,
-    @required this.disabledActiveTrackColor,
-    @required this.disabledInactiveTrackColor,
-    @required this.activeTickMarkColor,
-    @required this.inactiveTickMarkColor,
-    @required this.disabledActiveTickMarkColor,
-    @required this.disabledInactiveTickMarkColor,
-    @required this.thumbColor,
-    @required this.disabledThumbColor,
-    @required this.overlayColor,
-    @required this.valueIndicatorColor,
-    @required this.trackShape,
-    @required this.tickMarkShape,
-    @required this.thumbShape,
-    @required this.overlayShape,
-    @required this.valueIndicatorShape,
-    @required this.showValueIndicator,
-    @required this.valueIndicatorTextStyle,
-  }) : assert(trackHeight != null),
```

Comment:
Do you not need the asserts anymore because you have the factory methods? Are factory methods mandatory or could someone still use this const constructor?

Reply:
Someone can still use this constructor. The default Theme instances uses it. The asserts are no longer wanted because the SliderTheme now follows the same pattern as DialogTheme, FloatingActionButtonTheme, etc., where fields are allowed to be null, because the defaults are within the component itself. The component first checks for its own constructor values, then any intermediate values, including ones from the theme. If these are all null it falls back to the material default defined in its own class.